src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,475 @@
+---
+title: "Basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad2 code examples."
+toc: true
+--- 
```

Comment:
Welcome to the Flutter layout codelab, where you learn to
lay out a Flutter UI without downloading or installing
Dart or Flutter!

Estimated time to complete this codelab: 40-60 minutes.  (Or whatever seems reasonable.)